tests/unit/remoteworker/test_remote_worker.py

```diff
@@ -0,0 +1,143 @@
+        self.challenge_phase_pk = 1
+        self.data = {"test": "data"}
+        self.headers = {"Authorization": "Token test_token"}
+
+    def make_request_url(self):
+        return "/test/url"
+
+    def get_message_from_sqs_queue_url(self, queue_name):
+        return "/api/jobs/challenge/queues/{}/".format(queue_name)
+
+    def delete_message_from_sqs_queue_url(self, queue_name, receipt_handle):
+        return "/api/jobs/queues/{}/receipt/{}/".format(queue_name, receipt_handle)
+
+    def get_submission_by_pk_url(self, submission_pk):
+        return "/api/jobs/submission/{}".format(submission_pk)
+
+    def get_challenge_phases_by_challenge_pk_url(self, challenge_pk):
+        return "/api/challenges/{}/phases/".format(challenge_pk)
+
+    def get_challenge_by_queue_name_url(self, queue_name):
+        return "/api/challenges/challenge/queues/{}/".format(queue_name)
+
+    def get_challenge_phase_by_pk_url(self, challenge_pk, challenge_phase_pk):
+        return "/api/challenges/challenge/{}/challenge_phase/{}".format(challenge_pk, challenge_phase_pk)
+
+    def update_submission_data_url(self, challenge_pk):
+        return "/api/jobs/challenge/{}/update_submission/".format(challenge_pk)
+
+
+@mock.patch("scripts.workers.remote_submission_worker.AUTH_TOKEN", "test_token")
+@mock.patch("scripts.workers.remote_submission_worker.requests")
+class MakeRequestTestClass(BaseTestClass):
+    def setUp(self):
+        super(MakeRequestTestClass, self).setUp()
+        self.url = super(MakeRequestTestClass, self).make_request_url()
+
+    def test_make_request_get(self, mock_make_request):
+        make_request(self.url, "GET")
+        mock_make_request.get.assert_called_with(url=self.url, headers=self.headers)
+
+    def test_make_request_put(self, mock_make_request):
+        make_request(self.url, "PUT", data=self.data)
+        mock_make_request.put.assert_called_with(url=self.url, headers=self.headers, data=self.data)
+
+    def test_make_request_patch(self, mock_make_request):
+        make_request(self.url, "PATCH", data=self.data)
+        mock_make_request.patch.assert_called_with(url=self.url, headers=self.headers, data=self.data)
+
+
+@mock.patch("scripts.workers.remote_submission_worker.QUEUE_NAME", "evalai_submission_queue")
+@mock.patch("scripts.workers.remote_submission_worker.return_url_per_environment")
+@mock.patch("scripts.workers.remote_submission_worker.make_request")
+class APICallsTestClass(BaseTestClass):
+
+    def test_get_message_from_sqs_queue(self, mock_make_request, mock_url):
+        url = self.get_message_from_sqs_queue_url("evalai_submission_queue")
+        get_message_from_sqs_queue()
+        mock_url.assert_called_with(url)
+        url = mock_url(url)
+        mock_make_request.assert_called_with(url, "GET")
+
+    def test_delete_message_from_sqs_queue(self, mock_make_request, mock_url):
+        test_receipt_handle = "MbZj6wDWli+JvwwJaBV+3dcjk2YW2vA3+STFFljTM8tJJg6HRG6PYSasuWXPJB+Cw"
```

Comment:
@KhalidRmb What is the expiry date of this receipt handle? I think it will change after some time and the test will start failing. Did you already check that it is constant every time?

Reply:
@RishabhJain2018 It works since I'm mocking out the make_request function. Since I'm unit testing it separately anyhow.

On second thought, I think it would be better if I add a test case using mock_ecs alone. Will do it 👍